dm/worker/upgrade.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package worker
+
+import (
+	"encoding/binary"
+	"os"
+	"strconv"
+
+	"github.com/pingcap/errors"
+	"github.com/syndtr/goleveldb/leveldb"
+
+	"github.com/pingcap/dm/pkg/log"
+)
+
+const (
+	// The current internal version of DM-worker used when upgrading from an older version, and it's different from the release version.
+	// NOTE: +1 when an incompatible problem is introduced.
+	currentWorkerVersion uint64 = 1
```

Comment:
can we use our version like `1.0.1`? we can only support support to upgrade between official version

Reply:
Then after each incompatible problem introduced, a new version needs to be released?
If not, users upgrade from an older version to _latest_ will encounter the problem.
By the way, TiDB is also using this strategy, see https://github.com/pingcap/tidb/blob/cf5f42b8ece0fad2b3f93c6753747b392f33f4c8/session/bootstrap.go#L313.